### PR TITLE
Feature/email captcha backend

### PR DESCRIPTION
**Pull request by pushpendranirwan**
Kindly review for email and captcha.

### CODE REVIEW

File: vfox-restServer/src/main/java/com/vfoexchange/restServer/controller/UserController.java [hunk 1]
```diff
@@ -12,18 +15,24 @@
...
     @Autowired
     private UserService userService;
     @Autowired
     private EmailServices emailServices;
 
-    @Autowired
-    Environment environment;
 
     private static Logger LOGGER = LoggerFactory.getLogger(UserController.class);
     /*
```
Comment (ajayvfox) [inline vfox-restServer/src/main/java/com/vfoexchange/restServer/controller/UserController.java]: LOGGER should be the first line on all classes.

File: vfox-restServer/src/main/java/com/vfoexchange/restServer/controller/UserController.java [hunk 2]
```diff
@@ -43,6 +52,13 @@ Method for adding new user(user can be advisor or admin)
             return responseEntity;
         }
         try {
+            Mail mail = new Mail();
+            mail.setFrom(environment.getProperty("spring.mail.username"));
+            mail.setTo(userDto.getUsername());
+            mail.setSubject("Verify Your Email Address");
+            mail.setContent(AppUtil.getMailBody(AppUtil.getURL(AppUtil.getEncodedString(userDto.getUsername()))));
+
+            emailServices.sendMail(mail);
```
Comment (ajayvfox) [inline vfox-restServer/src/main/java/com/vfoexchange/restServer/controller/UserController.java:58]: Delete line 62.

File: vfox-restServer/src/main/java/com/vfoexchange/restServer/coreApp/DbConfig.java [hunk 3]
```diff
@@ -42,5 +42,23 @@ public JdbcTemplate jdbcTemplate(DataSource dataSource) {
...
 
+    @Bean
+    public JavaMailSender javaMailService() {
+        JavaMailSenderImpl mailSender = new JavaMailSenderImpl();
+
+        Properties mailProperties = new Properties();
+        mailProperties.put("mail.smtp.auth", env.getProperty("spring.mail.properties.mail.smtp.auth"));
+        mailProperties.put("mail.smtp.starttls.enable",  env.getProperty("spring.mail.properties.mail.smtp.starttls.enable"));
+        mailProperties.put("mail.smtp.socketFactory.class", "javax.net.ssl.SSLSocketFactory");
+
```
Comment (ajayvfox) [inline vfox-restServer/src/main/java/com/vfoexchange/restServer/coreApp/DbConfig.java]: Always remove extra blank lines. Just one blank line please.

File: vfox-restServer/src/main/resources/application.properties [hunk 4]
```diff
@@ -1,4 +1,4 @@
-server.port=8080
+server.port=8081
```
Comment (ajayvfox) [inline vfox-restServer/src/main/resources/application.properties]: Why change of port?

File: vfox-restServer/src/main/resources/dbScript/baseV0.1.sql [hunk 5]
```diff
@@ -26,7 +26,7 @@ CREATE TABLE IF NOT EXISTS `vfox`.`User` (
   `CreatedAt` DATETIME NOT NULL,
   `UpdatedBy` INT NOT NULL,
   `UpdatedAt` DATETIME NOT NULL,
-  PRIMARY KEY (`Id`),
+  PRIMARY KEY (`Id`,`UserName`),
```
Comment (ajayvfox) [inline vfox-restServer/src/main/resources/dbScript/baseV0.1.sql]: Why are we changing user key?

**Review by ajayvfox — COMMENTED** [review]
Added some initial comments. Please review.

File: vfox-restServer/src/main/java/com/vfoexchange/restServer/controller/UserController.java (hunk 1 above)
Comment (pushpendranirwan) [reply vfox-restServer/src/main/java/com/vfoexchange/restServer/controller/UserController.java]: Done !

File: vfox-restServer/src/main/java/com/vfoexchange/restServer/controller/UserController.java (hunk 2 above)
Comment (pushpendranirwan) [reply vfox-restServer/src/main/java/com/vfoexchange/restServer/controller/UserController.java:58]: Done.

File: vfox-restServer/src/main/java/com/vfoexchange/restServer/coreApp/DbConfig.java (hunk 3 above)
Comment (pushpendranirwan) [reply vfox-restServer/src/main/java/com/vfoexchange/restServer/coreApp/DbConfig.java]: Done.

File: vfox-restServer/src/main/resources/application.properties (hunk 4 above)
Comment (pushpendranirwan) [reply vfox-restServer/src/main/resources/application.properties]: Port has been changed.

File: vfox-restServer/src/main/resources/dbScript/baseV0.1.sql (hunk 5 above)
Comment (pushpendranirwan) [reply vfox-restServer/src/main/resources/dbScript/baseV0.1.sql]: Handle Duplicate Records in User table.

Comment (ajayvfox) [reply vfox-restServer/src/main/resources/dbScript/baseV0.1.sql]: Why can't we add 'Unique Index' on 'UserName' instead of making it part of Primary key.

**Review by ajayvfox — COMMENTED** [review]
Please check the comment regarding changes to 'primary key'.

File: vfox-restServer/src/main/resources/dbScript/baseV0.1.sql (hunk 5 above)
Comment (pushpendranirwan) [reply vfox-restServer/src/main/resources/dbScript/baseV0.1.sql]: Indexes are used to retrieve data from the database very fast.

Comment (pushpendranirwan) [reply vfox-restServer/src/main/resources/dbScript/baseV0.1.sql]: Change has been done.

**Review by ajayvfox — APPROVED** [review]
LGTM

**Pull request merged**